### PR TITLE
Swtiched logging from file to inmemory

### PR DESCRIPTION
**Pull request by suhasj**
@HaoK @divega 

I switched from logging the messages in memory from writing it to a file since the this was causing the tests to be flaky

### CODE REVIEW

**Comment by HaoK** [comment]
You can get rid of the file lock entirely now no?  Each test should be getting it's own logger so there is no contention

**Comment by suhasj** [comment]
I read that List is not thread safe. Would it happen that during the tests there might be read and write simultaneously. This was the reason why it was failing for files.

**Comment by HaoK** [comment]
So each test manager should be getting it's own logger Instance if that is not the case already you should fix that

On Jan 18, 2015, at 2:47 PM, "Suhas Joshi" <notifications@github.com<mailto:notifications@github.com>> wrote:


I read that List is not thread safe. Would it happen that during the tests there might be read and write simultaneously. This was the reason why it was failing for files.

-

**Comment by suhasj** [comment]
oh yes it is getting it's own ILogger instance. I'll remove the lock and check the result. Hopefully it works fine with the in memory implementation

**Comment by suhasj** [comment]
@HaoK  Removed the lock property and it tests are passing on my machine. Tried multiple times

File: test/Shared/IdentityResultAssert.cs [hunk 1]
```diff
@@ -60,15 +59,12 @@ public static void VerifyRoleManagerSuccessLog(ILogger logger, string methodName
         }
         private static void VerifySuccessLog(ILogger logger, string className, string methodName, string id, string userOrRole = "user")
         {
-            if (logger is TestFileLogger)
+            if (logger is TestLogger)
```
Comment (HaoK) [inline test/Shared/IdentityResultAssert.cs]: nit: combine as and is check so you can just test against null

File: test/Shared/IdentityResultAssert.cs [hunk 2]
```diff
@@ -78,13 +74,10 @@ private static void VerifySuccessLog(ILogger logger, string className, string me
 
         public static void VerifyLogMessage(ILogger logger, string expectedLog)
         {
-            if (logger is TestFileLogger)
+            if (logger is TestLogger)
```
Comment (HaoK) [inline test/Shared/IdentityResultAssert.cs]: same thing combine perhaps rename var to testLogger too

File: test/Shared/TestLogger.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Identity.Test
+{
+    public class TestLogger : ILogger
+    {
+        public IList<string> LogMessages { get; private set; } = new List<string>();
+
+        public TestLogger(string name)
```
Comment (HaoK) [inline test/Shared/TestLogger.cs]: name is unused.  remove

**Comment by suhasj** [comment]
@HaoK updated with feedback

File: test/Shared/IdentityResultAssert.cs [hunk 4]
```diff
@@ -60,15 +59,11 @@ public static void VerifyRoleManagerSuccessLog(ILogger logger, string methodName
         }
         private static void VerifySuccessLog(ILogger logger, string className, string methodName, string id, string userOrRole = "user")
         {
-            if (logger is TestFileLogger)
+            TestLogger testlogger;
```
Comment (HaoK) [inline test/Shared/IdentityResultAssert.cs]: its more readable putting the as in the declaration rather than the if

**Comment by HaoK** [comment]
:shipit: after you move the as's out of the ifs

**Comment by suhasj** [comment]
sure thanks @HaoK

**Pull request closed without merging**